plugins/gstperf.c

```diff
@@ -198,13 +246,86 @@ gst_perf_get_property (GObject * object, guint property_id,
   }
 }
 
+static gboolean
+gst_perf_update_bps (void *data)
+{
+  guint buffer_current_idx;
+  GstPerf *perf;
+  guint byte_count;
+  gdouble bps, mean_bps;
+
+  g_return_val_if_fail (data, FALSE);
+
+  perf = GST_PERF (data);
+
+  g_mutex_lock (&perf->byte_count_mutex);
+  byte_count = perf->byte_count;
+  perf->byte_count = G_GUINT64_CONSTANT (0);
+  g_mutex_unlock (&perf->byte_count_mutex);
+
+  g_mutex_lock (&perf->mean_bps_mutex);
+  mean_bps = perf->mean_bps;
+  g_mutex_unlock (&perf->mean_bps_mutex);
+
+  /* Calculate bits per second */
+  bps = byte_count * GST_PERF_BITS_PER_BYTE / (perf->bps_interval / 1000.0);
```

Comment:
Can you get rid of the magic 1000? Use a local constant with a descriptive name.